src/mcts/params.cc

```diff
@@ -55,6 +58,104 @@ FillEmptyHistory EncodeHistoryFill(std::string history_fill) {
   return FillEmptyHistory::NO;
 }
 
+ContemptPerspective EncodeContemptPerspective(std::string perspective) {
+  if (perspective == "sidetomove") return ContemptPerspective::STM;
+  if (perspective == "white") return ContemptPerspective::WHITE;
+  if (perspective == "black") return ContemptPerspective::BLACK;
+  assert(perspective == "none");
+  return ContemptPerspective::NONE;
+}
+
+float GetContempt(std::string name, std::string contempt_str,
+                  float uci_rating_adv) {
+  float contempt = uci_rating_adv;
+  for (auto& entry : StrSplit(contempt_str, ",")) {
+    auto parts = StrSplit(entry, "=");
+    if (parts.size() == 1) {
+      try {
+        contempt = std::stof(parts[0]);
+      } catch (std::exception& e) {
+        throw Exception("Invalid default contempt: " + entry);
+      }
+    } else if (parts.size() == 2) {
+      if (std::search(name.begin(), name.end(), parts[0].begin(),
```

Comment:
This is to check that parts[x] is a substring of name, right? (case insensitive).
(are you sure that you need a substring rather than a full match)

Reply:
IIRC the original reasoning was to catch something like "Dragon" and "Komodo Dragon".